draft-sury-toorop-dnsop-server-cookies.md

```diff
@@ -245,6 +257,60 @@ Notice that Client-IP is used for hash generation even though it's not included
 in the cookie value itself. Client-IP can be either 4 bytes for IPv4 or 16
 bytes for IPv6.
 
+# Updating the Server Secret {#rollingSecret}
+
+All servers in an anycast must be able to verify the Server Cookies constructed
+by all other servers in that anycast set at all times.  Therefore it is vital
+that the Server Secret is shared among all servers before it us used to
+generate Server Cookies.
+
+Also, to maximize maintaining established relationships between clients and
+servers, an old Server Secret should be valid for verification purposes for a
+specific period.
+
+To facilitate this, deployment of a new Server Secret MUST be done in three
+stages:
+
+Stage 1
+: The new Server Secret is deployed on all the servers in an anycast set by
+  the operator.
+
+> Each server learns the new Server Secret, but keeps using the previous Server
+  Secret to generate Server Cookies.
+
+> Server Cookies constructed with the both the new Server Secret and with
+  the previous Server Secret are considered valid when verifying.
+
+> After stage 1 completed, all the servers in the anycast set have learned the
+  new Server Secret, and can verify Server Cookies constructed with it, but keep
+  generator Server Cookies with the old Server Secret.
+
+Stage 2
+: This stage starts either by explicit signalling from the operator, or after
+  a configurable period after which it is reasonable to assume that the Server
+  Cookie is present on all members from the anycast set.  A default value for
+  such a configurable time could be 5 minutes.
+
+> When entering Stage 2, servers start generating Server Cookies with the new
+  Server Secret. The previous Server Secret is not yet removed/forgotten about.
+
+> Server Cookies constructed with the both the new Server Secret and with
+  the previous Server Secret are considered valid when verifying.
+
+Stage 3
+: This stage starts either by explicit signalling from the operator, or after
+  a configurable period after which it is reasonable to assume that most
+  clients have learned the new Server Secret.
+
+> We RECOMMEND this period to be the longest TTL in the zones served by the
+  server plus half an hour, but it SHOULD be at least longer than the period
+  clients are allowed to use the same Server Cookie, which SHOULD be half an
+  hour, see (#timestampField).
```

Comment:
However, this recommendation should be kept.

Reply:
addressed in commit @9cf8dbc